tools/build-webview

```diff
@@ -30,90 +32,108 @@ err() {
 }
 
 ################################################################################
-# Parameters and environment
+# Environment and globals
 ################################################################################
 
-# chdir to the current git repo's root
-ROOT="$(git rev-parse --show-toplevel)"
-cd "$ROOT"
+# The project root (or at least the current git repo's root), for absolutizing
+# paths.
+root="$(git rev-parse --show-toplevel)"
+readonly root
+
+# This is our local staging directory; once it's assembled, it'll be synced over
+# to $dest.
+readonly staging="$root"/src/webview/static
+if [ ! -d "$staging" ]; then
+    err "cannot find asset staging directory '$staging'";
+fi
```

Comment:
I think this bit belongs more naturally around where it was, shortly before where it's used. One reason is that this isn't the right spot for this error check; if the command is `tools/build-webview --help`, or ill-formed, the right behavior is to exit with a syntax message, which means doing CLI parsing first.

Another is just that this feels closely tied to the details of how the build and sync work, so a natural thing to want to consult when reading that. That isn't a huge deal for this variable, because it's pretty simple; in the other PR, I felt it more strongly on the `z_rsync` function definition, which was also up here separated from the code that uses it.

Reply:
Will do. (In the very near future, I'd like to pull the staging directory out of the middle of the source tree, and let `src/webview/static` be just another source directory for it. Once that's done, it would make some sense to be able to specify an arbitrary staging directory anyway – at least for testing – so there's another reason to keep it in that general area!)

Reply:
SGTM!